### PR TITLE
Add package starpu

### PR DESCRIPTION
**Pull request by fpruvost**
add fxt, simgrid and starpu packages

### CODE REVIEW

File: var/spack/repos/builtin/packages/simgrid/package.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    version('3.13', '8ace1684972a01429d5f1c5db8966709',
+            url='http://gforge.inria.fr/frs/download.php/file/35817/SimGrid-3.13.tar.gz')
+    version('3.12', 'd73faaf81d7a9eb0d309cfd72532c5f1',
+            url='http://gforge.inria.fr/frs/download.php/file/35215/SimGrid-3.12.tar.gz')
+    version('3.11', '358ed81042bd283348604eb1beb80224',
+            url='http://gforge.inria.fr/frs/download.php/file/33683/SimGrid-3.11.tar.gz')
+    version('3.10', 'a345ad07e37539d54390f817b7271de7',
+            url='http://gforge.inria.fr/frs/download.php/file/33124/SimGrid-3.10.tar.gz')
+    version('master', git='https://scm.gforge.inria.fr/anonscm/git/simgrid/simgrid.git')
+    version('starpumpi', git='https://scm.gforge.inria.fr/anonscm/git/simgrid/simgrid.git', branch='starpumpi')
```
Comment (fpruvost) [inline var/spack/repos/builtin/packages/simgrid/package.py]: ok, I'm waiting for the constraints to add to the starpu package

Comment (sthibaul) [inline var/spack/repos/builtin/packages/simgrid/package.py]: Well, I have to say I don't know which versions work with which. Yesterday I made sure again that the trunk works with 3.11 to 3.15, but I haven't tested the 1.2 branch, let alone older starpu versions. I'm pretty sure starpu 1.1 won't work with the latest simgrid for instance, though.

**Comment by fpruvost** [comment]
@stanisic @sthibaul ok for merge ?

**Comment by sthibaul** [comment]
Well, I asked a question about whether debug variants were already defined, for the rest it's ok for me.

**Comment by fpruvost** [comment]
@sthibaul ok.
Concerning the variant debug, do you think it is necessary to handle it in a specific way it for the starpu stack ?
It seems that now in spack we can now handle this debug variant generically, see
- http://spack.readthedocs.io/en/latest/features.html#custom-versions-configurations
- http://spack.readthedocs.io/en/latest/getting_started.html#manual-compiler-configuration

**Comment by sthibaul** [comment]
I have no idea. I'm just saying that perhaps the variant declaration was forgotten, since it seems all other variants are explicitly declared with variant() before being used with spec.satisifies. But again, if it's already predeclared by spack, then there is no bug, I'm not saying anything particular should be done. I was just surprised by the missing declaration

**Comment by fpruvost** [comment]
yes it was on purpose no to declare the variant debug. Let's try to manage this kind of feature with the spack native support.

**Pull request merged**